Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -757,7 +782,7 @@ private static void CommitRepo(IRepository repository, ArtifactDeploymentInfo zi
             // We need to make to call respository.Commit() since deployment flow expects at
             // least 1 commit in the IRepository. Even though there is no repo per se in this
             // scenario, deployment pipeline still generates a NullRepository
-            repository.Commit(zipDeploymentInfo.Message, zipDeploymentInfo.Author, zipDeploymentInfo.AuthorEmail);
+            repository.Commit(zipDeploymentInfo.Message, zipDeploymentInfo.Author, zipDeploymentInfo.AuthorEmail, zipDeploymentInfo.FixedDeploymentId);
```

Comment:
Can we have commitId as an optional parameter for NullRepository constructor and instantiate NullRepository accordingly in ArtifactDeploymentInfo.GetRepository()?

Reply:
Yeah, it'll work. Will update.